src/Plutarch/Extra/ExchangeRate.hs

```diff
@@ -0,0 +1,85 @@
+module Plutarch.Extra.ExchangeRate (
+    (:>),
+
+    -- * Exchange-rate conversions
+    exchangeFromTruncate,
+    exchangeToTruncate,
+    exchangeFrom,
+    exchangeTo,
+) where
+
+import GHC.TypeLits (Symbol)
+
+import Plutarch.Extra.Applicative (ppure)
+import Plutarch.Extra.Rational (divRational, divTruncate, mulRational, mulTruncate)
+import Plutarch.Extra.Tagged (PTagged)
+
+{- | Represents an exchange from a to b.
+     Let's say 1.00 ADA is worth 2.00 USD, then @ADA ':>' USD@ ought to be
+     represented as 2.00.
+
+   @since 3.8.0
+-}
+data (:>) (a :: Symbol) (b :: Symbol)
+
+{- | Exchange from one currency to another, truncating the result
+
+ @since 3.8.0
+-}
+exchangeFromTruncate ::
+    forall (a :: Symbol) (b :: Symbol) (s :: S).
+    Term
+        s
+        ( PTagged (a :> b) PRational :--> PTagged a PInteger
+            :--> PTagged b PInteger
+        )
+exchangeFromTruncate =
+    phoistAcyclic $
+        plam $ \ex x -> ppure #$ mulTruncate # pto ex # pto x
+
+{- | Exchange from  one currency to another, truncating the result
+ (inverse direction)
+
+ @since 3.8.0
+-}
+exchangeToTruncate ::
+    forall (a :: Symbol) (b :: Symbol) (s :: S).
+    Term
+        s
+        ( PTagged (a :> b) PRational :--> PTagged b PInteger
+            :--> PTagged a PInteger
+        )
+exchangeToTruncate =
+    phoistAcyclic $
+        plam $ \ex x -> ppure #$ divTruncate # pto ex # pto x
```

Comment:
I'm not sure I follow why we need both. They seem to only differ in variable naming in their signatures. Does `:<` imply a sizing relationship?

Reply:
As an example: `exchangeFromTruncate` takes an exchange rate from `1.00 ADA :> 2.00 USD` and an `2.00 ADA` value, and returns `4.00 USD`. 

`echangeToTruncate` takes an exchange rate `1.00 ADA :> 2.00 USD` and a `4.00 USD` value, and returns `2.00 ADA`. 

Am I missing something else?